support/src/main/java/org/akvo/flow/support/domain/SendConversation.kt

```diff
@@ -57,30 +55,27 @@ class SendConversation @Inject constructor(
     }
 
     private suspend fun findConversation(parameters: Map<String, Any>): Conversation? {
+        val refId = parameters[PARAM_REF_ID] as Int
         //load page 0
         var page = 0
         var conversationsResult = restApi.fetchConversations(page)
         var conversations = conversationsResult.conversations
-        var totalCount = conversationsResult.totalCount
-        var countLoaded = conversations?.size ?: 0
+        val pageCount = conversationsResult.pageCount
 
-        val refId = parameters[PARAM_REF_ID] as Int
         var conversation = findConversation(conversations, refId)
         if (conversation != null) {
             return conversation
         } else {
```

Comment:
Isnt it possible to have all this within the `do ... while()`?

Reply:
done!